src/airtable/project_summary.js

```diff
@@ -67,6 +84,56 @@ const remove = async (ids) => {
   })
 }
 
+const addUUID = async (proposals) => {
+  const groupedProposals = {}
+
+  // Grouping proposal by project name
+  proposals.forEach((proposal) => {
+    const key = proposal['Project Name']
+
+    if (!groupedProposals[key]) {
+      groupedProposals[key] = [proposal]
+    } else {
+      groupedProposals[key].push(proposal)
+    }
+  })
+
+  // Add UUID if not available
+  for (const key in groupedProposals) {
+    const proposalsCollection = groupedProposals[key]
+
+    const proposalWithUUID = proposalsCollection.find(
+      (x) => x.UUID !== undefined
+    )
+
+    if (proposalWithUUID) {
+      const uuid = proposalWithUUID.UUID
+
+      proposalsCollection.forEach(async (proposal) => {
+        if (proposal.UUID === undefined) {
+          await updatedRecordById(proposal.RecordId, { UUID: uuid })
+        }
+      })
+    } else {
+      const uuid = uuidv4()
```

Comment:
You can remove the duplicate code here.
Something like:
`const uuid = proposalWithUUID ? proposalWithUUID.UUID : uuidv4()`